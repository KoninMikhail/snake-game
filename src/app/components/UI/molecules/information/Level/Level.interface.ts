export interface ILevel {
    upText: string;
    downText: string;
}
