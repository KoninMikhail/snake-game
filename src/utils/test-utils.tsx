import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
