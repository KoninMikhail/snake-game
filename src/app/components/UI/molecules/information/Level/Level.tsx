import {
    StyledLevel,
    StyledLevelCounter,
    StyledLevelLabel,
} from '@ui/molecules/information/Level/style/StyledLevel';
import { Text } from '@ui/atoms/typography/body/Text/Text';
import { ExtraLargeCaption } from '@ui/atoms/typography/caption/ExtraLargeCaption/ExtraLargeCaption';
import useAppSelector from '@hooks/useAppSelector/useAppSelector';
import { gameLevelSelector } from '@store/slices/game.slice';
import { ILevel } from '@ui/molecules/information/Level/Level.interface';

/**
 * @name Level
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for display showLevel;
 */
export const Level = ({ upText, downText }: ILevel) => {
    const currentLevel = useAppSelector(gameLevelSelector);
    return (
        <>
            <StyledLevel>
                <StyledLevelCounter>
                    <ExtraLargeCaption>{currentLevel}</ExtraLargeCaption>
                </StyledLevelCounter>
                <StyledLevelLabel>
                    <Text>{upText}</Text>
                    <Text>{downText}</Text>
                </StyledLevelLabel>
            </StyledLevel>
        </>
    );
};
