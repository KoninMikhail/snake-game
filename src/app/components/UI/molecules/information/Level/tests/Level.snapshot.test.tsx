import { Level } from '../Level';
import { GameData } from '@store/slices/game.slice';
import { renderWithProviders } from '@utils/test-utils';

describe('Level', () => {
    const downText = 'level';
    const upText = 'current';

    it('equal snapshot', () => {
        const initialState = {
            preloadedState: { game: { level: 5 } as GameData },
        };
        const { container } = renderWithProviders(
            <Level downText={downText} upText={upText} />,
            initialState
        );

        expect(container).toMatchSnapshot();
    });
});
