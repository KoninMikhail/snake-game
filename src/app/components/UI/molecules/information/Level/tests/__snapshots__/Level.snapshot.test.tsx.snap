// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level equal snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: min-content auto;
  gap: 1rem;
}

.c1 {
  display: grid;
}

.c3 {
  display: grid;
  grid-template-columns: 1fr;
}

.c4 {
  color: white;
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
}

.c2 {
  font-size: 2.3em;
  font-weight: normal;
}

@media (min-width:768px) {
  .c4 {
    font-size: 16px;
  }
}

@media (min-width:1200px) {
  .c4 {
    font-size: 1.1vw;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.3em;
  }
}

@media (min-width:1200px) {
  .c2 {
    font-size: 2em;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        5
      </span>
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        current
      </span>
      <span
        class="c4"
      >
        level
      </span>
    </div>
  </div>
</div>
`;
